pkg/controller/s3/bucket/versioningConfig.go

```diff
@@ -54,12 +53,8 @@ func (in *VersioningConfigurationClient) LateInitialize(ctx context.Context, buc
 		bucket.Spec.ForProvider.VersioningConfiguration = &v1beta1.VersioningConfiguration{}
 		config = bucket.Spec.ForProvider.VersioningConfiguration
 	}
-	if len(external.Status) != 0 { // By default Status is the string ""
-		config.Status = aws.String(string(external.Status))
-	}
-	if len(external.MFADelete) != 0 { // By default MFADelete is the string ""
-		config.MFADelete = aws.String(string(external.MFADelete))
-	}
+	config.Status = aws.LateInitializeStringPtr(config.Status, aws.String(string(external.Status)))
+	config.MFADelete = aws.LateInitializeStringPtr(config.MFADelete, aws.String(string(external.MFADelete)))
 	return nil
```

Comment:
Minor nitpick: I believe the default values of external.Status and external.MFADelete is empty strings `""`, so with this implementation, we are always late-initializing the config to be a struct with those values, even though that's not what the user provided, or what the external representation is. I think the current implementation works, but it just leaves some ambiguous values.

Reply:
I checked the implementation and `aws.String(string(external.Status))` returns `nil` if the given string is `""`. So, we don't give any value if the returned string is `""`